provider/provider.go

```diff
@@ -6,4 +6,6 @@ import (
 
 type Provider interface {
 	NotifyMergeRequestCreated(mergeRequest model.MergeRequestInfo) error
+	ReactToMessage(mergeRequest model.MergeRequestInfo, reactionRule model.ReactionRule) error
```

Comment:
I don't think it does because this method is agnostic. It can react to any operation configured by the `reactionRule` parameter. e.g.

```json
[{
  "action": "merged",
  "reaction": "tada",
}]
```

Reply:
As agreed on a call, let's refactor it to have the reactions just on the slack provider.

To help with that we will have more specific methods on the provider interface for merge requests opened, approved, closed, etc.

The merge request actions are described [here](https://docs.gitlab.com/ee/user/project/integrations/webhook_events.html#merge-request-events).